ffi/athcon/include/athcon/athcon.h

```diff
@@ -820,6 +820,27 @@ extern "C"
   struct athcon_vm *athcon_create_example_vm(void);
 #endif
 
+  typedef struct athcon_bytes_t
+  {
+    const uint8_t* ptr;
+    size_t size;
+  } athcon_bytes;
+
+  void athcon_free_bytes(athcon_bytes* v);
+
+  athcon_bytes* athcon_encode_tx(
+    const athcon_address* principal_account,
+    athcon_address* template_addresss,
+    uint8_t* method,
+    size_t method_size,
+    uint64_t nonce,
+    uint8_t* args,
+    size_t args_size
+  );
+
+  athcon_bytes* athcon_encode_tx_spawn(const athcon_bytes32* pubkey);
```

Comment:
What do we do about error handling here? Does the caller just assume that an empty return value indicates error?

Reply:
Scale encoding is infallible.